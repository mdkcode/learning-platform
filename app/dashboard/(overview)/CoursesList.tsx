import { getCourseVideos } from "@/app/api/data";
import { CourseCard } from "@/app/ui/dashboard/CourseCard";

export interface CourseProps {
  id: number;
  title: string;
  url: string;
  description: string;
}

export default async function CoursesList({
  searchQuery,
}: {
  searchQuery?: string;
}) {
  const videos = await getCourseVideos(searchQuery);
  return (
    <div className="flex gap-5 flex-wrap">
      {videos?.map((video: CourseProps) => (
        <CourseCard key={video.id} {...video} />
      ))}
    </div>
  );
}
