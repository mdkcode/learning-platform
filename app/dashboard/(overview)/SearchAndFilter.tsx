export default function SearchAndFilter() {
  return (
    <div>
      <input
        type="search"
        placeholder="Search courses..."
        className="w-full max-w-md p-3 mb-4 border border-gray-300 rounded-lg 
                   focus:outline-none focus:ring-1 transition-all duration-300"
      />
    </div>
  );
}
