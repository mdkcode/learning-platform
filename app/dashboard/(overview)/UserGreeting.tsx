"use client";
import { useSession } from "next-auth/react";

export function UserGreeting() {
  const { data: session } = useSession();

  return (
    <h1 className="my-5 text-3xl font-bold">
      Welcome, {session?.user?.name ?? "Guest"}
    </h1>
  );
}
