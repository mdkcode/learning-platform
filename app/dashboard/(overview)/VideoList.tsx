"use client";
import { CourseCard } from "@/app/ui/dashboard/CourseCard";
import { useEffect, useState } from "react";

export interface CourseVideoProps {
  id: number;
  title: string;
  url: string;
  description: string;
}

const VideoList = () => {
  const [videos, setVideos] = useState<CourseVideoProps[]>([]);

  useEffect(() => {
    const fetchVideos = async () => {
      try {
        const response = await fetch("/courseData.json");
        const data = await response.json();
        setVideos(data);
      } catch (error) {
        console.error("Error fetching videos:", error);
      }
    };
    fetchVideos();
  }, []);

  return (
    <div>
      {videos.map((video) => (
        <CourseCard key={video.id} {...video} />
      ))}
    </div>
  );
};

export default VideoList;
