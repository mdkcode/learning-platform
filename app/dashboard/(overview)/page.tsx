export default function Page() {
  return (
    <main>
      <h1 className="my-5 text-3xl font-bold">Welcome</h1>
      <h1 className="text-2xl">Take a look at the available courses</h1>
    </main>
  );
}
