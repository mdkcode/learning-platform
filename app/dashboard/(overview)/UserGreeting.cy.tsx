import React from 'react'
import { UserGreeting } from './UserGreeting'

describe('<UserGreeting />', () => {
  it('renders', () => {
    // see: https://on.cypress.io/mounting-react
    cy.mount(<UserGreeting />)
  })
})