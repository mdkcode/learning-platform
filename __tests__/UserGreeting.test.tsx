import { render } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { UserGreeting } from "@/app/dashboard/(overview)/UserGreeting";
import { useSession } from "next-auth/react";

vi.mock("next-auth/react", () => ({
  useSession: vi.fn(),
}));

describe("UserGreeting", () => {
  it("matches snapshot with user name", () => {
    (useSession as ReturnType<typeof vi.fn>).mockReturnValue({
      data: {
        user: {
          name: "Alice",
        },
        expires: "future-date",
      },
    });

    const { container } = render(<UserGreeting />);
    expect(container).toMatchSnapshot();
  });

  it("matches snapshot without user name", () => {
    (useSession as ReturnType<typeof vi.fn>).mockReturnValue({
      data: null,
    });

    const { container } = render(<UserGreeting />);
    expect(container).toMatchSnapshot();
  });
});
