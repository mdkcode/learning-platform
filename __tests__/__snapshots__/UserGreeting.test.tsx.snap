// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserGreeting > matches snapshot with user name 1`] = `
<div>
  <h1
    class="my-5 text-3xl font-bold"
  >
    Welcome, 
    Alice
  </h1>
</div>
`;

exports[`UserGreeting > matches snapshot without user name 1`] = `
<div>
  <h1
    class="my-5 text-3xl font-bold"
  >
    Welcome, 
  </h1>
</div>
`;
